Dockerfile.build

```diff
@@ -0,0 +1,26 @@
+FROM fedora:26
+
+COPY bdcs-deps.dnf /etc/yum.repos.d/bdcs-deps.repo
+RUN dnf -y install sudo cabal-install python-toml python-nose-parameterized \
+                   gobject-introspection-devel ghc-*weldr* \
+                   ghc-haskell-gi{,-devel}-0.20.3-1.fc26.weldr.1 \
+                   libgit2-glib{,-devel}-0.26.0-2.fc26.weldr.1 && \
+    dnf clean all
+
+RUN cabal update && cabal install happy hlint
+
+RUN mkdir /bdcs-cli
+WORKDIR /bdcs-cli
+
+# install the build dependencies first so we can cache this layer
+COPY BDCSCli.cabal .
+RUN cabal update && cabal install --dependencies-only --enable-tests --force-reinstall
```

Comment:
Is the install really needed? It may mask problems with the ghc rpm packages.
I think we want to switch to depending on just the rpms so that we can be sure everything is packaged correctly.

Reply:
Hmmm, do we know if we have all necessary requirements as RPMs ? Should I just figure out their names and `dnf install` together with the other packages in this Dockerfile ?

Reply:
@bcl 

```
cabal: Encountered missing dependencies:
aeson -any,
aeson-pretty -any,
cond -any,
gi-ggit -any,
gi-gio -any,
gi-glib -any,
hspec -any,
htoml -any,
lens -any,
lens-aeson -any,
monad-loops -any,
semver -any,
split -any,
temporary -any,
unordered-containers -any,
wreq -any
```

This is after I install ghc-aeson and pretty much everything else. My command was `for i in `cabal install --dependencies-only --dry-run | grep - | grep -v "In order" | \
              grep -v basement | grep -v overloading | sed -r 's|-[0-9]+(\.[0-9]+)*||'`; do \
        echo ghc-$i; done | xargs dnf -y install && dnf clean all
`

any ideas ?

Reply:
Here's what I came up with for my docker development image (I currently can't get my F26 desktop to work because some of these conflict with desktop packages I have installed) - https://github.com/bcl/docker-bdcs-hs

Reply:
OK, that worked. I needed the -devel counterpars of everything plus --enable-testing. Not sure if I need --enable-coverage with --dry-run though.